scripts/firedrake-install

```diff
@@ -780,14 +780,14 @@ def get_petsc_options(minimal=False):
             petsc_options.add("--CFLAGS=-Wno-implicit-function-declaration")
             if Version(clt["version"]) >= Version("15"):
                 # CLT >= 15 requires legacy linking behaviour
-                petsc_options.add("--LDFLAGS=-Wl,-ld_classic")
+                petsc_options.add("--LDFLAGS=-Wl,-ld_classic,-dead_strip_dylibs")
```

Comment:
Please add a comment pointing out that this option has been cargo culted.

Reply:
done.